examples/demo_2d_swe.py

```diff
@@ -0,0 +1,54 @@
+
+from __future__ import division  # Get proper divison
+
+import math
+import random
+
+import numpy as np
+
+
+from firedrake import *
+
+from flooddrake import *
+
+""" demo file for simple 2d shallow water equations """
+
+
+# Meshsize
+
+n = 10
+mesh = UnitSquareMesh(n, n)
+
+# mixed function space
+X = FunctionSpace(mesh, "DG", 1)
+Y = FunctionSpace(mesh, "DG", 1)
+Z = FunctionSpace(mesh, "DG", 1)
+V = X * Y * Z
+
+
+# for slope limiter
+XCG = FunctionSpace(mesh, "CG", 1)
+YCG = FunctionSpace(mesh, "CG", 1)
+ZCG = FunctionSpace(mesh, "CG", 1)
+VCG = XCG * YCG * ZCG
+
+
+# setup free surface depth
+g = interpolate(Expression(
+    ['pow(x[0]-0.5,2) + pow(x[1]-0.5,2)< 0.05 ? 1 : (pow(x[0]-0.5,2) + pow(x[1]-0.5,2)< 0.05 ? -1.0 : 0.8)', 0, 0]), V)
```

Comment:
Another option for using UFL interpolation is to interpolate directly onto the subfunction:
```python
g = Function(V)
g.sub(0).interpolate(...)
```
Since your second and third components are zero, you just need to interpolate onto the first one.